src/main/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClient.java

```diff
@@ -233,6 +233,7 @@ public class AmazonDynamoDBLockClient implements Runnable, Closeable {
     private final boolean holdLockOnServiceUnavailable;
     private final String ownerName;
     private final ConcurrentHashMap<String, LockItem> locks;
+    private final ConcurrentHashMap<String, LockItem> notMyLocks;
```

Comment:
Do we need to purge entries from the map eventually?

Reply:
If an app takes out leases with constantly changing names, then this can grow unboundedly.
If the lease names that it attempts to acquire are limited, then not.

Reply:
Updated to limit entries to 100K